manager/logbroker/broker.go

```diff
@@ -15,58 +19,152 @@ import (
 // Log subscriptions are pushed to the work nodes by creating log subscsription
 // tasks. As such, the LogBroker also acts as an orchestrator of these tasks.
 type LogBroker struct {
-	store       *store.MemoryStore
-	broadcaster *events.Broadcaster
-	nodes       []*api.Node
-	mu          sync.RWMutex
+	broadcaster   *events.Broadcaster
+	subscriptions *events.Broadcaster
+
+	stopped chan struct{}
 }
 
-func NewLogBroker() *LogBroker {
+// New initializes and returns a new LogBroker
+func New() *LogBroker {
 	return &LogBroker{
-		broadcaster: events.NewBroadcaster(),
+		broadcaster:   events.NewBroadcaster(),
+		subscriptions: events.NewBroadcaster(),
```

Comment:
Same with these - they need to be closed when the log broker stops.

Take care when closing them that there are no sinks still attached to the broadcasters; otherwise it can deadlock. We have a wrapper in `manager/state/watch` that we use to protect against this, which you may find useful here.